source/common/event/user_space_file_event_impl.cc

```diff
@@ -0,0 +1,67 @@
+#include "common/event/user_space_file_event_impl.h"
+
+#include <cstdint>
+
+#include "common/common/assert.h"
+
+namespace Envoy {
+namespace Event {
+
+void DefaultEventListener::onEventEnabled(uint32_t enabled_events) {
+  enabled_events_ = enabled_events;
+  // Clear ephemeral events to align with FileEventImpl::setEnable().
+  ephermal_events_ = 0;
+}
+
+void DefaultEventListener::onEventActivated(uint32_t activated_events) {
+  // Event owner should not activate any event which is disabled.
+  // Also see onEventEnabled which clears ephemeral events.
+  // The overall prevents callback on disabled events.
+  ASSERT((activated_events & ~enabled_events_) == 0);
```

Comment:
I was incorrect about this.  Network::ConnectionImpl operation, specifically operations related to handling of high watermarks and readDisable do require calling activate in cases where the requested events are a superset of enabled_events_.  This ASSERT will trigger when these classes are used with Network::ConnectionImpl.

Reply:
Thanks for the update!
My impression is that activate could be set in order to trigger `consumerWantsToRead()`. Not sure we are referring to the same case.

Reply:
That's correct, it is possible for there to be a call to activate(read) while the enabled mask does not contain read when read_buffer being above the high-watermark; consumerWantsToRead returns true despite the connection being read disabled.

Reply:
Thank you for the explanation! I add a TODO here see if we can assert when exclude this known case.
Brain stop working now